<?xml version="1.0" encoding="UTF-8"?>
<testng-results ignored="0" total="3" passed="2" failed="1" skipped="0">
  <reporter-output>
    <line>
      <![CDATA[b]]>
    </line>
    <line>
      <![CDATA[c]]>
    </line>
  </reporter-output>
  <suite started-at="2023-01-12T17:46:01 IST" name="Default suite" finished-at="2023-01-12T17:46:08 IST" duration-ms="7532">
    <groups>
      <group name="regression">
        <method signature="GroupExecutionEx1.a()[pri:0, instance:testng.practice.ListenerImplement@33990a0c]" name="a" class="testng.practice.GroupExecutionEx1"/>
        <method signature="GroupExecutionEx1.c()[pri:0, instance:testng.practice.ListenerImplement@33990a0c]" name="c" class="testng.practice.GroupExecutionEx1"/>
      </group> <!-- regression -->
      <group name="smoke">
        <method signature="GroupExecutionEx1.a()[pri:0, instance:testng.practice.ListenerImplement@33990a0c]" name="a" class="testng.practice.GroupExecutionEx1"/>
        <method signature="GroupExecutionEx1.b()[pri:0, instance:testng.practice.ListenerImplement@33990a0c]" name="b" class="testng.practice.GroupExecutionEx1"/>
      </group> <!-- smoke -->
      <group name="both">
        <method signature="GroupExecutionEx1.a()[pri:0, instance:testng.practice.ListenerImplement@33990a0c]" name="a" class="testng.practice.GroupExecutionEx1"/>
      </group> <!-- both -->
    </groups>
    <test started-at="2023-01-12T17:46:01 IST" name="Default test" finished-at="2023-01-12T17:46:08 IST" duration-ms="7532">
      <class name="testng.practice.ListenerImplement">
        <test-method signature="a()[pri:0, instance:testng.practice.ListenerImplement@33990a0c]" started-at="2023-01-12T17:46:01 IST" name="a" finished-at="2023-01-12T17:46:08 IST" duration-ms="6814" status="FAIL">
          <exception class="java.lang.AssertionError">
            <message>
              <![CDATA[null]]>
            </message>
            <full-stacktrace>
              <![CDATA[java.lang.AssertionError: null
at org.testng.Assert.fail(Assert.java:99)
at org.testng.Assert.fail(Assert.java:106)
at testng.practice.GroupExecutionEx1.a(GroupExecutionEx1.java:32)
at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
at java.base/java.lang.reflect.Method.invoke(Method.java:564)
at org.testng.internal.MethodInvocationHelper.invokeMethod(MethodInvocationHelper.java:133)
at org.testng.internal.TestInvoker.invokeMethod(TestInvoker.java:598)
at org.testng.internal.TestInvoker.invokeTestMethod(TestInvoker.java:173)
at org.testng.internal.MethodRunner.runInSequence(MethodRunner.java:46)
at org.testng.internal.TestInvoker$MethodInvocationAgent.invoke(TestInvoker.java:824)
at org.testng.internal.TestInvoker.invokeTestMethods(TestInvoker.java:146)
at org.testng.internal.TestMethodWorker.invokeTestMethods(TestMethodWorker.java:146)
at org.testng.internal.TestMethodWorker.run(TestMethodWorker.java:128)
at java.base/java.util.ArrayList.forEach(ArrayList.java:1511)
at org.testng.TestRunner.privateRun(TestRunner.java:794)
at org.testng.TestRunner.run(TestRunner.java:596)
at org.testng.SuiteRunner.runTest(SuiteRunner.java:377)
at org.testng.SuiteRunner.runSequentially(SuiteRunner.java:371)
at org.testng.SuiteRunner.privateRun(SuiteRunner.java:332)
at org.testng.SuiteRunner.run(SuiteRunner.java:276)
at org.testng.SuiteRunnerWorker.runSuite(SuiteRunnerWorker.java:53)
at org.testng.SuiteRunnerWorker.run(SuiteRunnerWorker.java:96)
at org.testng.TestNG.runSuitesSequentially(TestNG.java:1212)
at org.testng.TestNG.runSuitesLocally(TestNG.java:1134)
at org.testng.TestNG.runSuites(TestNG.java:1063)
at org.testng.TestNG.run(TestNG.java:1031)
at org.testng.remote.AbstractRemoteTestNG.run(AbstractRemoteTestNG.java:115)
at org.testng.remote.RemoteTestNG.initAndRun(RemoteTestNG.java:251)
at org.testng.remote.RemoteTestNG.main(RemoteTestNG.java:77)
]]>
            </full-stacktrace>
          </exception> <!-- java.lang.AssertionError -->
          <reporter-output>
          </reporter-output>
        </test-method> <!-- a -->
        <test-method signature="b()[pri:0, instance:testng.practice.ListenerImplement@33990a0c]" started-at="2023-01-12T17:46:08 IST" name="b" finished-at="2023-01-12T17:46:08 IST" duration-ms="2" status="PASS">
          <reporter-output>
            <line>
              <![CDATA[b]]>
            </line>
          </reporter-output>
        </test-method> <!-- b -->
        <test-method signature="c()[pri:0, instance:testng.practice.ListenerImplement@33990a0c]" started-at="2023-01-12T17:46:08 IST" name="c" finished-at="2023-01-12T17:46:08 IST" duration-ms="681" status="PASS">
          <reporter-output>
            <line>
              <![CDATA[c]]>
            </line>
          </reporter-output>
        </test-method> <!-- c -->
      </class> <!-- testng.practice.ListenerImplement -->
    </test> <!-- Default test -->
  </suite> <!-- Default suite -->
</testng-results>
